### PR TITLE
Highlight with mouse should not close dropdown

### PR DESCRIPTION
**Pull request by matbmapspeople**
The scope of this task is to prevent dropdown to close when highlighting content inside input field, while mouse click is released outside the input field.

**How does it look like now:**

https://github.com/MapsPeople/web-ui/assets/56337591/08c51df1-b3bb-448a-b47a-29ee4a63887e

My implementation proposition:

Inside `checkForClickOutside()`:
- I created couple of consts: for listening clicks inside hostElement or inside canvas (map), also const that checks if anything inside an input is highlighted with mouse.
- Then I made if checks.

I tried to be descriptive with JSDocs, but let me know if you have any questions.

**My implementation in action:**

https://github.com/MapsPeople/web-ui/assets/56337591/85e96e8a-5772-49ef-8bfd-eab12d3d39b0

### CODE REVIEW

File: packages/components/src/components/combo-box/combo-box.tsx [hunk 1]
```diff
@@ -122,24 +123,72 @@ export class ComboBox {
...
+    svgButtonToggleDropdown!: SVGElement;
     clearButtonElement!: HTMLButtonElement;
     listElement!: HTMLElement;
 
     currentItemIndex = 0;
     selectedItemIndex = 0;
     highlightedItemClassName = 'list__item--highlighted';
 
+    /**
+     * If we click on the map (canvas) we want to empty selection, so this.open can be set to false.
```
Comment (ammapspeople) [inline packages/components/src/components/combo-box/combo-box.tsx:135]: What map are we referring to here?

File: packages/components/src/components/combo-box/combo-box.tsx [hunk 2]
```diff
@@ -122,24 +123,72 @@ export class ComboBox {
...
      */
     @Listen('click', { target: 'window' })
+    @Listen('mouseup', { target: 'window' })
     checkForClickOutside(ev) {
-        if (!this.hostElement.contains(ev.target) && this.noResultsMessage) {
-            this.clearFilter();
-            this.filterElement.value = this.selected[0]?.text;
-            this.open = false;
+        // Check if the event target is inside the input field or canvas
+        const clicksInsideBody = document.getElementsByTagName('body');
```
Comment (ammapspeople) [inline packages/components/src/components/combo-box/combo-box.tsx:155]: I don't think I understand the naming of this variable. `getElementsByTagName('body')` will return a collection of HTML elements (and how can multiple body elements exist in the document? Can't `document.body` be used instead).

How does that relate to something with clicks?

File: packages/components/src/components/combo-box/combo-box.tsx [hunk 3]
```diff
@@ -122,24 +123,72 @@ export class ComboBox {
...
+        const isTargetInsideInput = this.hostElement.contains(ev.target);
+        const isTargetInsideCanvas = clicksInsideBody[0].contains(ev.target);
```
Comment (ammapspeople) [inline packages/components/src/components/combo-box/combo-box.tsx:157]: What is "canvas" here? `clicksInsideBody[0]` refers to a `body` element.

File: packages/components/src/components/combo-box/combo-box.tsx [hunk 4]
```diff
@@ -122,24 +123,72 @@ export class ComboBox {
...
+
+        // Check if there is any selected text inside the input field
+        const selection = window.getSelection().toString();
+        const hasSelectedText = selection.length > 0;
```
Comment (ammapspeople) [inline packages/components/src/components/combo-box/combo-box.tsx:161]:
```suggestion
        const hasSelectedText = window.getSelection().toString().length > 0;
```

File: packages/components/src/components/combo-box/combo-box.tsx [hunk 5]
```diff
@@ -122,24 +123,72 @@ export class ComboBox {
...
+
+        const inputInFocus = this.isComboboxInputFieldInFocus('menu__combobox');
+
+        // If we click outside an input field but the text was highlihted with the mouse, dropdown will remain opened.
```
Comment (ammapspeople) [inline packages/components/src/components/combo-box/combo-box.tsx:165]:
```suggestion
        // If we click outside an input field but the text was highlighted with the mouse, dropdown will remain opened.
```

File: packages/components/src/components/combo-box/combo-box.tsx [hunk 6]
```diff
@@ -122,24 +123,72 @@ export class ComboBox {
...
+     * @returns {boolean}
+     */
+    isComboboxInputFieldInFocus(inputFieldClassName: string): boolean {
+        const focusedElement = document.activeElement;
+
+        if (focusedElement.className === inputFieldClassName) {
+            return true;
+        } else {
+            return false;
         }
```
Comment (ammapspeople) [inline packages/components/src/components/combo-box/combo-box.tsx:192]:
```suggestion
        return focusedElement.className === inputFieldClassName;
```

**Comment by matbmapspeople** [comment]
Closed due to no more need to this complicated fix.

**Pull request closed without merging**